src/engine/ruleset/decoders/wazuh-agent/syscollector/syscollector-dbsync.yml

```diff
@@ -0,0 +1,108 @@
+# packages
+#  {"data":{"architecture":"x86_64","checksum":"0d7e29ef72b657c96512b43efa6637ad0431b2e5","description":"Tool to analyse BIOS DMI data","format":"rpm","groups":"Unspecified","install_time":"1607103514","item_id":"bbb009a26c1b5bc26c093e9fbcd9b01d0fd227d4","location":null,"multiarch":null,"name":"dmidecode","priority":null,"scan_time":"2023/03/14 15:09:43","size":222289,"source":null,"vendor":"CentOS","version":"1:3.2-6.el8"},"operation":"DELETED","type":"dbsync_packages"}
+# network_iface
+#  {"data":{"adapter":null,"checksum":"052a00a969cb2d54a97890b20ea9af156b74d42d","item_id":"d4aa8b01955e438235d586585492f3f5edceec1b","mac":"52:54:00:27:8b:50","mtu":1500,"name":"eth0","rx_bytes":510595,"rx_dropped":0,"rx_errors":0,"rx_packets":4522,"scan_time":"2023/03/10 03:27:12","state":"up","tx_bytes":616053,"tx_dropped":0,"tx_errors":0,"tx_packets":4186,"type":"ethernet"},"operation":"MODIFIED","type":"dbsync_network_iface"}
+# network_address
+#  {"data":{"address":"fe80::a00:27ff:fe0d:a1e5","broadcast":null,"checksum":"7bdb80f090fb4aa51756a4fd0382a1b08ee3e0be","dhcp":"unknown","iface":"eth1","item_id":"4a53b37f2fb0d3ef21c7ff8b5e1b9072b578b82f","metric":null,"netmask":"ffff:ffff:ffff:ffff::","proto":1,"scan_time":"2023/03/14 15:32:01"},"operation":"INSERTED","type":"dbsync_network_address"}
+# network_protocol
+# {"data":{"checksum":"ea17673e7422c0ab04c4f1f111a5828be8cd366a","dhcp":"unknown","gateway":"192.168.0.1|600","iface":"enp4s0","item_id":"9dff246584835755137820c975f034d089e90b6f","metric":" ","type":"ipv6"},"operation":"INSERTED","type":"dbsync_network_protocol"}
+# hotfixes
+# {"data":{"scan_time":"2021/10/29 14:26:24","hotfix":"KB123456","checksum":"abcdef0123456789"}, "type":"dbsync_hotfixes","operation":"MODIFIED"}
+
+
+
+normalize:
+  - check: +ef_exists/~json.data.address OR +ef_exists/~json.data.mac OR +ef_exists/~json.data.os_version OR +ef_exists/~json.data.release
+    map:
+      - ~do_update_kvdb: true
+
+  - check: +s_ne/~json.data.hostname/ OR +s_ne/~json.data.os_name/ OR +s_ne/~json.data.os_platform/ OR +s_ne/~json.data.sysname/
+    map:
+      - ~do_update_kvdb: true
+
+  # Avoid updating when architecture field not from osinfo
+  - check: +s_ne/~json.data.architecture/ AND +s_eq/~json.type/osinfo
+    map:
+      - ~do_update_kvdb: true
+
+  - map:
+      - ~json.type: +s_replace/dbsync_/
+      - ~tmp.query: +s_concat/agent /$agent.id/ dbsync /$~json.type/ /$~json.operation/ /$~json.data
+      - ~tmp.query_response: +wdb_query/$~tmp.query
+      - ~json.data.checksum: +ef_delete
+
+  - check:
+      - ~tmp.query_response: +s_ne/
+    map:
+      - ~do_update_kvdb: false
+
+    # Mapping result fields to wazuh.syscollector
+  - check: +s_eq/~tmp.query_response/ AND +s_eq/~json.type/processes
```

Comment:
`wazuh.syscollector` must be updated if the query is successful. In that case, `~tmp.query_response` is not empty. Replace `+s_eq/~tmp.query_response/` with `+s_ne/~tmp.query_response/`

Reply:
indeed, I'll change and recheck that!